src/AI/COParser.java

```diff
@@ -0,0 +1,59 @@
+package AI;
+
+import java.util.ArrayList;
+
+import CommandingOfficers.Commander;
+import CommandingOfficers.DeployableCommander;
+import CommandingOfficers.RuinedCommander;
+import Engine.Army;
+import Units.Unit;
+
+/**
+ * Figures out which of your COs have a special sub-type you can glean more info from.
+ */
+public class COParser
+{
+  public final ArrayList<DeployableCommander> deployableCOs;
+  // Will necessarily be in the above list, but there may be more flavors later
+  public final ArrayList<RuinedCommander> dorCOs;
```

Comment:
I'm... not sure if I should ask.

Reply:
The Tabitha conversion, for example, won't be a "real" `RuinedCommander` - she uses Trilogy defense, has global charge, etc.